src/network.js

```diff
@@ -191,24 +240,36 @@ class Network {
   /**
    * Connects to another peer
    *
-   * @param {PeerId|Multiaddr} peer
-   * @param {Object} options
-   * @param {AbortSignal} options.abortSignal
+   * @param {PeerId|Multiaddr|Provider} peer
+   * @param {Object} [options]
+   * @param {AbortSignal} [options.signal]
    * @returns {Promise<Connection>}
    */
   async connectTo (peer, options) { // eslint-disable-line require-await
     if (!this._running) {
       throw new Error('network isn\'t running')
     }
 
-    return this.libp2p.dial(peer, options)
+    // TODO: Figure out inconsistency here.
```

Comment:
@vasco-santos could you please help me out understand what is expected here. Passed `peer` here comes from `libp2p.contentRouting.findProviders` which contains both peer id and multiaddr, but then `libp2p.dial` expects either `peerId` or `multiaddr` and attempts to turn it back into `Provider`.

I do not know all the ins and outs here, but clearly something makes wrong assumptions.

Reply:
Created an issue in libp2p https://github.com/libp2p/js-libp2p/issues/899